pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -661,7 +771,22 @@ def expval(self, observable, shot_range=None, bin_size=None):
                 return np.squeeze(np.mean(samples, axis=0))
 
             if observable.name in ["SparseHamiltonian"]:
-                CSR_SparseHamiltonian = observable.sparse_matrix().tocsr()
+                if not self._mpi:
+                    CSR_SparseHamiltonian = observable.sparse_matrix().tocsr()
+                    return self.measurements.expval(
+                        CSR_SparseHamiltonian.indptr,
+                        CSR_SparseHamiltonian.indices,
+                        CSR_SparseHamiltonian.data,
+                    )
```

Comment:
Since the return statement is the same for MPI and non-MPI, could we just use the if-else to define the `CSR_SparseHamiltonian` object (MPI, or nonMPI), and use the single return at the end?

Reply:
Done!